DQMOffline/Muon/src/MuonRecoAnalyzer.cc

```diff
@@ -332,76 +375,182 @@ void MuonRecoAnalyzer::GetRes( reco::TrackRef t1, reco::TrackRef t2, string par,
+  edm::Handle<DcsStatusCollection> dcsStatus;
+  bool fillBadLumi = false;
+  if(iEvent.getByToken(dcsStatusCollection_, dcsStatus) && dcsStatus.isValid()) {
+     for (auto const & dcsStatusItr : *dcsStatus) {
+       if (!dcsStatusItr.ready(DcsStatus::CSCp))   fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::CSCm))   fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::DT0))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::DTp))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::DTm))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::EBp))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::EBm))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::EEp))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::EEm))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::ESp))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::ESm))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::HBHEa))  fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::HBHEb))  fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::HBHEc))  fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::HF))     fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::HO))     fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::BPIX))   fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::FPIX))   fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::RPC))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::TIBTID)) fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::TOB))    fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::TECp))   fillBadLumi = true;
+       if (!dcsStatusItr.ready(DcsStatus::TECm))   fillBadLumi = true;
+       //if (!dcsStatusItr.ready(DcsStatus::CASTOR)) fillBadLumi = true;
+    }
+  }
 
   float res=0, pull=0;
   if(!muons.isValid()) return;
 
   for (edm::View<reco::Muon>::const_iterator muon = muons->begin(); muon != muons->end(); ++muon){
-      
+
+
+    //Needed for MVA soft muon
+    
+    reco::TrackRef gTrack = muon->globalTrack();
+    reco::TrackRef iTrack = muon->innerTrack();
+    reco::TrackRef oTrack = muon->outerTrack();
+    if ( iTrack.isNonnull() && oTrack.isNonnull() && gTrack.isNonnull() ) {
+      const reco::HitPattern gHits = gTrack->hitPattern();
+      const reco::HitPattern iHits = iTrack->hitPattern();
+      const reco::MuonQuality muonQuality = muon->combinedQuality();
+      int pvIndex = 0;   
+      math::XYZPoint refPoint;
+      if(doMVA) {
+         pvIndex = getPv(iTrack.index(), &(*vertex)); //HFDumpUtitilies
+         if (pvIndex > -1) {
+	   refPoint = vertex->at(pvIndex).position();
+         } else {
+	   if (&(*beamSpot)==NULL) {
+	     refPoint = beamSpot->position();
+	   } else {
+	     cout << "ERROR: No beam sport found!" << endl;
```

Comment:
This message is appearing e.g. in 20034.0 step3 logs (phase2 workflow). `cout` should not be used.

Also, I'm not sure I understand the logic that you try to assign the `refPoint` only if the `beamSpot` address is null...